docs/content/plugins/search/algolia.md

```diff
@@ -100,38 +100,94 @@ Where `<YOUR_APP_ID>` and `<YOUR_ADMIN_API_KEY>` are respectively the Applicatio
 
 Finally, in `medusa-config.js` add the following item into the `plugins` array:
 
-```jsx title=medusa-config.js
+```js title=medusa-config.js
 const plugins = [
   // ...
   {
     resolve: `medusa-plugin-algolia`,
     options: {
-      application_id: process.env.ALGOLIA_APP_ID,
-      admin_api_key: process.env.ALGOLIA_ADMIN_API_KEY,
+      applicationId: process.env.ALGOLIA_APP_ID,
+      adminApiKey: process.env.ALGOLIA_ADMIN_API_KEY,
       settings: {
-        products: {
-          searchableAttributes: ["title", "description"],
-          attributesToRetrieve: [
-            "id",
-            "title",
-            "description",
-            "handle",
-            "thumbnail",
-            "variants",
-            "variant_sku",
-            "options",
-            "collection_title",
-            "collection_handle",
-            "images",
-          ],
+        // index settings...
+      },
+    },
+  },
+]
+```
+
+### Index Settings
+
+Under the `settings` key of the plugin's options, you can add settings specific to each index. The settings are of the following format:
+
+```js
+const plugins = [
+  // ...
+  {
+    resolve: `medusa-plugin-algolia`,
+    options: {
+      // other options...
+    settings: {
+      indexName: {
+        indexSettings: {
+          searchableAttributes,
+          attributesToRetrieve,
         },
+        transformer,
       },
     },
+    },
   },
 ]
 ```
 
-The `searchableAttributes` are the attributes in a product that are searchable, and `attributesToRetrieve` are the attributes to retrieve for each product result. You’re free to make changes to these attributes as you see fit, but these are the recommended attributes.
+Where:
+
+- `indexName`: the name of the index to create in Algolia. For example, `products`. Its value is an object containing the following properties:
+  - `indexSettings`: an object that includes the following properties:
+    - `searchableAttributes`: an array of strings indicating the attributes in the product entity that can be searched.
+    - `attributesToRetrieve`: an array of strings indicating the attributes in the product entity that should be retrieved in the search results.
+  - `transformer`: an optional function that accepts a product as a parameter and returns an object to be indexed. This allows you to have more control over what you're indexing. For example, you can add details related to variants or custom relations, or you can filter out certain products.
```

Comment:
@olivermrbl can you please confirm my understanding of the `transform` function?

Reply:
Yes, this is correct. You can use it for any type of Medusa entity, you wish to index. But right now, we only supports products :)